src/GraphQL/TypeApi.hs

```diff
@@ -245,40 +250,47 @@ instance forall typeName interfaces fields m.
   buildResolver mHandler selectionSet = do
     -- First we run the actual handler function itself in IO.
     handler <- mHandler
-    -- We're evaluating an Object so we're collecting (name, Value)
-    -- pairs from runFields and build a GValue.Map with them.
-    r <- forM selectionSet $ \selection -> runFields @m @fields handler selection
-    pure $ GValue.toValue (GValue.mapFromList r)
+    -- We're evaluating an Object so we're collecting ObjectFields from
```

Comment:
FWIW I think we should catch duplicate-field errors on query validation, not query evaluation. But we can keep the code for now. Maybe add a TODO